### PR TITLE
migration js to ts

### PR DESCRIPTION
**Pull request by ayocodingit**
## Overview

- migration javascript to typescript

cc: @jabardigitalservice/jds-backend 

## Evidence
- title: migration javascript to typescript
- project: Digiteam
- participants: @firmanalamsyah580 @f1reFr0st @fimz17 @sandisunandar

### CODE REVIEW

**Comment by sandisunandar99** [comment]
mantap dong js -> ts 👍

File: template/message.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ interface bodyValid {
   }
 }
 
-const messageValid = (payload: bodyValid) => {
+const messageValid = (payload: bodyValidInterface) => {
```
Comment (firmanJS) [inline template/message.ts:18]: Mising return type data in function

Comment (ayocodingit) [reply template/message.ts]: thanks i forgot to return type data :smile: :+1:

**Review by firmanJS — APPROVED** [review]

File: capture/exception.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-Raven.config(Config.get('sentry.dsn'), {
-  environment: Config.get('node.env')
+raven.config(config.get('sentry.dsn'), {
+  environment: config.get('node.env')
 }).install()
 
-const captureException = (error: any): void => {
-  if (error.message !== 'body not valid') Raven.captureException(error)
+const captureException = (error: Error): void => {
+  if (error.message !== 'body not valid') raven.captureException(error)
```
Comment (setiadijoe) [inline capture/exception.ts:9]: I preferred no hardcode value

Comment (ayocodingit) [reply capture/exception.ts:9]: Any solution ?

Comment (setiadijoe) [reply capture/exception.ts:9]: why don't create a package to save any constants so when the constants change, just change that package?

Comment (ayocodingit) [reply capture/exception.ts:9]: Oh i see, save to config or how ?

File: capture/exception.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import raven from 'raven'
+import config from '../config'
+
+raven.config(config.get('sentry.dsn'), {
+  environment: config.get('node.env')
+}).install()
+
+const captureException = (error: Error): void => {
+  if (error.message !== 'body not valid') raven.captureException(error)
```
Comment (sandisunandar99) [inline capture/exception.ts:9]: maybe message 'body not valid` make collection like `lang` or `message.json` so, if sentralize config this msg can reusable 👍

Comment (setiadijoe) [reply capture/exception.ts:9]: this is what I mean

Comment (ayocodingit) [reply capture/exception.ts:9]: Nah itu wkwkw

Comment (sandisunandar99) [reply capture/exception.ts:9]: that's true boys,,, 😆

Comment (ayocodingit) [reply capture/exception.ts]: updated yes df25ea7

File: utils/verifySecretKey.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import config from '../config'
+
+const verifySecretKey = async (secret: string): Promise<void> => {
+  if (secret !== config.get('secret.key')) throw Error('Credential is invalid')
```
Comment (sandisunandar99) [inline utils/verifySecretKey.ts:4]: _Credential is invalid_ .. maybe this same with upper coment .. 👍

Comment (ayocodingit) [reply utils/verifySecretKey.ts]: update dude @sandisunandar99 :smile:

Comment (sandisunandar99) [reply utils/verifySecretKey.ts]: mantappss boys,, wkwkwk

**Comment by sandisunandar99** [comment]
I thnk no coment again .. L>G>T>M

**Review by sandisunandar99 — APPROVED** [review]

**Pull request merged**